tozny/resource_client_registration_token.go

```diff
@@ -0,0 +1,199 @@
+        Default:     true,
+        ForceNew:    true,
+      },
+      "one_time_use": {
+        Description: "Whether the token is only valid for registering a single client.",
+        Type:        schema.TypeBool,
+        Optional:    true,
+        Default:     false,
+        ForceNew:    true,
+      },
+      "client_credentials_filepath": {
+        Description: "The filepath to Tozny client credentials for the provider to use when provisioning this registration token.",
+        Type:        schema.TypeString,
+        Optional:    true,
+        ForceNew:    true,
+      },
+      "token": {
+        Description: "Client registration token.",
+        Type:        schema.TypeString,
+        Computed:    true,
+        ForceNew:    true,
+        Sensitive:   true,
+      },
+    },
+  }
+}
+
+func resourceClientRegistrationTokenCreate(ctx context.Context, d *schema.ResourceData, m interface{}) diag.Diagnostics {
+  var diags diag.Diagnostics
+  var err error
+
+  toznySDK := m.(*e3db.ToznySDKV3)
+
+  toznyClientCredentialsFilePath := d.Get("client_credentials_filepath").(string)
+
+  if toznyClientCredentialsFilePath != "" {
+    toznySDK, err = e3db.GetSDKV3(toznyClientCredentialsFilePath)
+    if err != nil {
+      return diag.FromErr(err)
+    }
+  }
+
+  account, err := toznySDK.Login(ctx, toznySDK.AccountUsername, toznySDK.AccountPassword, "password", toznySDK.APIEndpoint)
+
+  if err != nil {
+    return diag.FromErr(err)
+  }
+
+  tokenName := d.Get("name").(string)
+  tfAllowedTypes := d.Get("allowed_registration_client_types").([]interface{})
+  allowedTypes := []string{}
+  for _, tfAllowedType := range tfAllowedTypes {
+    allowedTypes = append(allowedTypes, tfAllowedType.(string))
+  }
+
+  createTokenResponse, err := toznySDK.CreateRegistrationToken(ctx, accountClient.CreateRegistrationTokenRequest{
+    AccountServiceToken: account.Token,
+    Name:                tokenName,
+    TokenPermissions: accountClient.TokenPermissions{
+      Enabled:      d.Get("enabled").(bool),
+      AllowedTypes: allowedTypes,
+    },
+  })
+
+  if err != nil {
+    return diag.FromErr(err)
+  }
+
+  clientRegistrationToken := createTokenResponse.Token
+  d.Set("token", clientRegistrationToken)
+
+  // Associate created token with Terraform state and signal success
+  d.SetId(fmt.Sprintf("%d%v", time.Now().Unix(), tokenName))
+
+  return diags
+}
+
+func resourceClientRegistrationTokenRead(ctx context.Context, d *schema.ResourceData, m interface{}) diag.Diagnostics {
+  var diags diag.Diagnostics
+  var err error
+
+  toznySDK := m.(*e3db.ToznySDKV3)
+
+  toznyClientCredentialsFilePath := d.Get("client_credentials_filepath").(string)
+
+  if toznyClientCredentialsFilePath != "" {
+    toznySDK, err = e3db.GetSDKV3(toznyClientCredentialsFilePath)
+    if err != nil {
+      return diag.FromErr(err)
+    }
+  }
+
+  account, err := toznySDK.Login(ctx, toznySDK.AccountUsername, toznySDK.AccountPassword, "password", toznySDK.APIEndpoint)
```

Comment:
This login is just to get the API token?

Reply:
yes